packages/core/admin/server/services/api-token.js

```diff
@@ -130,6 +130,28 @@ const create = async (attributes) => {
   return result;
 };
 
+/**
+ * @param {string|number} id
+ *
+ * @returns {Promise<ApiToken>}
+ */
+const regenerate = async (id) => {
+  const accessKey = crypto.randomBytes(128).toString('hex');
+
+  const apiToken = await strapi.query('admin::api-token').update({
```

Comment:
Do we assume the provided ID is valid here?

Reply:
(I get that we do have checks in the controller but what if we call this from another method? Should we delegate the checks to all the callers?)